pkg/api/vlabs/types.go

```diff
@@ -272,7 +272,7 @@ const (
 // KubernetesConfig contains the Kubernetes config structure, containing
 // Kubernetes specific configuration
 type KubernetesConfig struct {
-	KubernetesImageBase             string            `json:"kubernetesImageBase,omitempty"`
+	KubernetesImageBase             string            `json:"kubernetesImageBase,omitempty"` // Deprecated
```

Comment:
leaving this here will cause apimodels using this field to not return an error but the value of `KubernetesImageBase` will be ignored silently which might cause error confusion. Let's add a validation: if `KubernetesImageBase` is present, either:
- print a warning that `KubernetesImageBase` is deprecated and will be ignored, use `KubernetesSpecConfig .HyperkubeImageBase` instead or
- return an error explaining that `KubernetesImageBase` is deprecated, use `KubernetesSpecConfig .HyperkubeImageBase`

Reply:
Probably the first option if we don't want to break upgrade...